ckanext/multilang/templates/package/snippets/package_basic_fields.html

```diff
@@ -24,19 +24,67 @@
     {% set empty_extras = (limit or 1) - total_extras %}
     {% if empty_extras <= 0 %}{% set empty_extras = 1 %}{% endif %}
 
-    {% for extra in range(total_extras, total_extras + empty_extras) %}
+        {% for extra in range(total_extras, total_extras + empty_extras) %}
       {% set index = loop.index0 + (data.tags|count) %}
       {% set prefix = 'extra_tag__%d__' % index %}
-      {{ form.custom(
-        names=(prefix ~ 'key', prefix ~ 'value', prefix ~ 'deleted'),
-        id='field-extra-tag-%d' % index,
-        label=_('Localized Tag Name'),
-        values=('', '', False),
-        error=errors[prefix ~ 'key'] or errors[prefix ~ 'value']
-      ) }}
-    {% endfor %}
+
+<div class="form-group control-custom" style="margin-bottom: 0px;">
+	<label class="control-label" for="field-extra-tag-{{index + 1}}-key">{{ _('Localized Tag Name') }}</label>
+	<div class="controls editor">
+      <div style="display: flex">
+		<div class="row" style="width: 100%;">
+			<div class="col-md-6">
+				<div class="input-group">
+					<label for="field-extra-tag-{{index + 1}}-key" class="input-group-addon">{{ _('Key') }}</label>
+					<select style="padding: 0px;" index="{{ index }}" class="form-control" id="field-extra-tag-{{index + 1}}-key" type="text" name="extra_tag__{{index}}__key" value="" placeholder="">
+                    </select>
+					</div>
+				</div>
+				<div class="col-md-6">
+					<div class="input-group">
+						<label for="field-extra-tag-{{ index }}-value" class="input-group-addon">{{ _('Value') }}</label>
+						<input class="form-control" id="field-extra-tag-{{ index }}-value" type="text" name="extra_tag__{{ index }}__value" value="" placeholder="">
+						</div>
+					</div>
+				</div>
+			</div>
+        </div>
+    </div>
+  </div>
+{% endfor %}
   </div>
 {% endif %}
+<script>
+  document.addEventListener("DOMContentLoaded", function(){
+    $("#field-tags").change(function () {
+        let available_tags = document.getElementById('field-tags').value;
+        available_tags = available_tags.split(',');
+        const tags = available_tags.filter(function (item, i) {
+          const element = document.getElementById(`field-extra-tag-${+i + 1}-key`);
+          const dom_availability = element ? !element.value && true : !element;
+          return dom_availability;
+        });
+
+        const options = tags.map(function (tag) {
+          return(`<option value="${tag}">${tag}</option>`);
+        }).join('');
+
+        var i;
+        for (i = (+available_tags.length - +tags.length); i < +available_tags.length + 1; i++) {
+          const id = `field-extra-tag-${i}-key`;
+          const element = document.getElementById(id);
+          if (element){
+            if (options){
+              element.innerHTML = `<option value="">---------------</option>${options}`;
```

Comment:
Replacing HTML in JS just does not seem right.
Isn't there any way to create proper HTML in the first place?
If not, please add a reason as comment in the code.

Reply:
@etj, I left a label on this as a WIP cause I could not find a better way to change add HTML to the DOM, I could just take out the javascript, it was just an enhancement (not so much related to the issue) I added to enable a user to choose tags key value (for internationalization) already added to the multi-select widget without retyping it, the only way I could do it was by adding/removing DOM element using javascript.